server/http_endpoints_test.go

```diff
@@ -611,7 +612,11 @@ func TestEndpoint_Sequence(t *testing.T) {
 			"resolve":       "resolve_first",
 			"resolve_gamma": "resolve_gamma_first",
 			"last":          "resolve,resolve_gamma",
-		}},
+		}, 6},
+		{"multiple request uses", "/multiple-request-uses", test.Header{}, "", log{"r2": "r1", "default": "r2,r1"}, 3},
+		{"multiple proxy uses", "/multiple-proxy-uses", test.Header{}, "", log{"r2": "p1", "default": "r2,p1"}, 3},
+		{"multiple sequence uses", "/multiple-sequence-uses", test.Header{}, "", log{"r2": "r1", "r4": "r2", "r3": "r2", "default": "r4,r3"}, 5},
```

Comment:
Maybe we should add extra tests for the last two cases which calls a delayed backend and the http client will cancel its request, this should show up the possible blocks mentioned above.